### ISSUE TITLE
Unlooted NPCs despawn/decay too quickly

### ISSUE THREAD

**Issue by eTzmNcbkrng** [chromiecraft/chromiecraft#195]
##### ISSUE: FACTION SIDE:

Both
![Alliance](https://user-images.githubusercontent.com/1884642/108204869-3a88d100-711c-11eb-8179-e1b9b73ed450.png)
![Horde](https://user-images.githubusercontent.com/1884642/108204991-63a96180-711c-11eb-9ed9-138c233070a6.png)

##### CONTENT PHASE:

1-19

##### SMALL DESCRIPTION:

All killed NPCs despawn too quickly, looted and unlooted.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Unable to find any hard numbers/evidence for what the original times were but the current config is definitely too fast.

##### CURRENT BEHAVIOUR:

From src\server\worldserver\worldserver.conf.dist
```
Corpse.Decay.NORMAL    = 60
Corpse.Decay.RARE      = 300
Corpse.Decay.ELITE     = 300
Corpse.Decay.RAREELITE = 300
Corpse.Decay.WORLDBOSS = 3600

#
#    Rate.Corpse.Decay.Looted
#        Description: Multiplier for Corpse.Decay.* to configure how long creature corpses stay
#                     after they have been looted.
#         Default:    0.5

Rate.Corpse.Decay.Looted = 0.5
```

This indicates that normal NPCs decay after 60 seconds or 30 seconds after being looted.
_Incidentally, I was unable to replicate these numbers in my testing. I killed two boars at exactly the same time and looted only one, both despawned 60 seconds later._

60 seconds is much too fast a decay rate for unlooted NPCs.
On numerous occasions I have pulled multiple NPCs and the dead ones have already started despawning before I've finished combat.
Suggest a decay rate of at least 3 minutes for unlooted NPCs.

##### STEPS TO REPRODUCE THE PROBLEM:

There are a few places this can be tested anywhere with tightly packed NPCs that flee.
Skull Rock in the North East of Durotar is a good spot or anywhere there is Murlocs.

Step 1 Go to 19,17 in Ghostlands
Step 2 Start pulling groups of Murlocs
Step 3 For me, I was only able to kill 5 Murlocs before the first ones started despawning.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/a32310772487a7a547b1a239dfaf0967b19f7493

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by YunataSavior** [comment]
From my experience with Classic WOW, it should be 5 minutes.

**Comment by Zoidwaffle** [comment]
The issue is discussed here: https://github.com/dalaranwow/dalaran-wow/issues/971

On my own server I also increased the time the corpses remain, simply because "it felt right" but I never bothered to investigate it further (before now).

But the mechanics sound like they work differently, so making it work in the exact same way isn't possible without a major rewrite? However, from what I get from the link above, it would be "pretty Blizz like" to set `Corpse.Decay.NORMAL = 300` and call it a day.

**Comment by Stifler82** [comment]
@eTzmNcbkrng @YunataSavior @Zoidwaffle 

So, let's start from the concept that "this" is not a Bug!
Here they report: "Bugs".
This is a: "Setting" (Intended).

And above all, it's Blizz-Like, (but outside of that), because I don't care about Blizz-Like, I prefer to do things with "Common Sense" and not because they are "Blizz-Like".

So..
I think the settings are not clear to you, so I will try to explain how they work. 
```
Corpse.Decay.NORMAL    = 60
Corpse.Decay.RARE      = 300
Corpse.Decay.ELITE     = 300
Corpse.Decay.RAREELITE = 300
Corpse.Decay.WORLDBOSS = 3600
```
**These are the times expressed in milliseconds.**


**This setting instead..**
```
#
#    Rate.Corpse.Decay.Looted
#        Description: Multiplier for Corpse.Decay.* to configure how long creature corpses stay
#                     after they have been looted.
#         Default:    0.5

Rate.Corpse.Decay.Looted = 0.5
```
**is a multiplier.**

What does it mean?
It means that if I take this setting:
```
#
#    Rate.Corpse.Decay.Looted
#        Description: Multiplier for Corpse.Decay.* to configure how long creature corpses stay
#                     after they have been looted.
#         Default:    0.5

Rate.Corpse.Decay.Looted = 0.5
```
and I modify it like this:
```
#
#    Rate.Corpse.Decay.Looted
#        Description: Multiplier for Corpse.Decay.* to configure how long creature corpses stay
#                     after they have been looted.
#         Default:    0.5

Rate.Corpse.Decay.Looted = 1
```
These values will become like this:
```
Corpse.Decay.NORMAL    = 60 * 2 = 120
Corpse.Decay.RARE      = 300 * 2 = 600
Corpse.Decay.ELITE     = 300 * 2 = 600
Corpse.Decay.RAREELITE = 300 * 2 = 600
Corpse.Decay.WORLDBOSS = 3600 * 2 = 7200
```

Now, as in my video below I demonstrate that the despawn takes place after 1 minute whether the mob has been looted or not, my question is this:

Why should we increase this value?
60 seconds: Aren't they enough to loot a Mob?

If we consider that a "Click" corresponds to (approximately) 500 milliseconds
in 60 seconds you can make at least 120 clicks

So, my question is: **Why should we change this parameter?**

**Video Test:** [Unlooted NPCs despawn](https://youtu.be/zldounIuirk)

So, In conclusion:
- I'm waiting for your answer to my question.
- I close the report because it is not a BUG, and next time you post things like this in the "Suggestions" section just in case..

**Comment by Zoidwaffle** [comment]
OK, so I don't have access to change labels or anything here - so I can't change it to "suggestion". But yes, this is not a bug, this is indeed a suggestion.

60s is not much - in case you died, you are pretty much guaranteed you don't get to loot anything. Which seems a bit unfair... So I think "common sense" and "Blizz-like" matches pretty well by increasing it. Perhaps 5m is too much, but 1m is on the low side (and that's my opinion and why I changed it on my own server).

The time expressed above is not in milliseconds as you wrote, but thanks for the explanation anyway.

**Comment by Stifler82** [comment]
> OK, so I don't have access to change labels or anything here - so I can't change it to "suggestion". But yes, this is not a bug, this is indeed a suggestion.
> 
> The time expressed above is not in milliseconds as you wrote, but thanks for the explanation anyway.

@Zoidwaffle 
yes you're right I wrote a bullshit!
ahahahah
I meant in seconds

**Comment by eTzmNcbkrng** [comment]
> So, my question is: **Why should we change this parameter?**

Because 1 minute is not long enough to loot mobs when you have packs aggroing.
Or, lets say you pull a pack of 5 mobs, kill 2 then you die, corpse run back and the ones you killed are gone. Very frustrating.

Having mobs despawn before you have time to loot them is not "Common Sense".

>
>    Corpse.Decay.NORMAL    = 60
>    Corpse.Decay.RARE      = 300
>    Corpse.Decay.ELITE     = 300
>    Corpse.Decay.RAREELITE = 300
>    Corpse.Decay.WORLDBOSS = 3600

>These are the times expressed in milliseconds.

I think you're mistaken here, these are not milliseconds, if they are, what's the point?
Why have World Bosses decay in 3.6 seconds, normal mobs decay in 6/100 of a second and everything else in 3/10 of a second.

They're seconds.

And yes, I understand how multiplying by fractions work.

**Comment by illfated** [comment]
Just as a comparison, here is the relevant part from TrinityCore `worldserver.conf.dist` (source file):
```
#    Corpse.Decay.NORMAL
#    Corpse.Decay.RARE
#    Corpse.Decay.ELITE
#    Corpse.Decay.RAREELITE
#    Corpse.Decay.WORLDBOSS
#        Description: Time (in seconds) until creature corpse will decay if not looted or skinned.
#        Default:     300  - (5 Minutes, Corpse.Decay.NORMAL)
#                     300  - (5 Minutes, Corpse.Decay.RARE)
#                     300  - (5 Minutes, Corpse.Decay.ELITE)
#                     300  - (5 Minutes, Corpse.Decay.RAREELITE)
#                     3600 - (1 Hour, Corpse.Decay.WORLDBOSS)

Corpse.Decay.NORMAL    = 300
Corpse.Decay.RARE      = 300
Corpse.Decay.ELITE     = 300
Corpse.Decay.RAREELITE = 300
Corpse.Decay.WORLDBOSS = 3600

#
#    Rate.Corpse.Decay.Looted
#        Description: Multiplier for Corpse.Decay.* to configure how long creature corpses stay
#                     after they have been looted.
#         Default:    0.5

Rate.Corpse.Decay.Looted = 0.5
```

**Comment by Stifler82** [comment]
> Just as a comparison, here is the relevant part from TrinityCore `worldserver.conf.dist` (source file):
> 
> ```
> #    Corpse.Decay.NORMAL
> #    Corpse.Decay.RARE
> #    Corpse.Decay.ELITE
> #    Corpse.Decay.RAREELITE
> #    Corpse.Decay.WORLDBOSS
> #        Description: Time (in seconds) until creature corpse will decay if not looted or skinned.
> #        Default:     300  - (5 Minutes, Corpse.Decay.NORMAL)
> #                     300  - (5 Minutes, Corpse.Decay.RARE)
> #                     300  - (5 Minutes, Corpse.Decay.ELITE)
> #                     300  - (5 Minutes, Corpse.Decay.RAREELITE)
> #                     3600 - (1 Hour, Corpse.Decay.WORLDBOSS)
> 
> Corpse.Decay.NORMAL    = 300
> Corpse.Decay.RARE      = 300
> Corpse.Decay.ELITE     = 300
> Corpse.Decay.RAREELITE = 300
> Corpse.Decay.WORLDBOSS = 3600
> 
> #
> #    Rate.Corpse.Decay.Looted
> #        Description: Multiplier for Corpse.Decay.* to configure how long creature corpses stay
> #                     after they have been looted.
> #         Default:    0.5
> 
> Rate.Corpse.Decay.Looted = 0.5
> ```

I understand, but with all the respect I have for you illfated, I don't care what TC does or what Blizzard does.
It's not that: if TC makes a decision, then we as Monkeys who mimic what humans do, have to do what TC does.

if we put 300 as TC did, this will mean that the mobs will despawn after 5 minutes, after which they will respawn after another 5 minutes, and this will mean that EVERY player will have to wait 10 minutes to kill that creature again.

**Comment by illfated** [comment]
Fair enough, not a problem. My posted content was simply meant to provide any of the affected players with some external information, in case they want to do more research before returning with more solid evidence, based on source code, blue posts and factual observance.

**Comment by Stifler82** [comment]
> Fair enough, not a problem. My posted content was simply meant to provide any of the affected players with some external information, in case they want to do more research before returning with more solid evidence, based on source code, blue posts and factual observance.

@illfated 
I hope you know, that I didn't mean to be aggressive and that I love you so much? ❤😜

**Comment by illfated** [comment]
Thanks, that is very nice of you. I have grown used to how some people write their posts, so I don't necessarily react much to it, at least not from users I have read a lot of comment posts from. Mind you, I can sometimes lose the overview of how much is affected by some of the issues, so in those cases I do take note of any follow-up comments from other users.

**Comment by Stifler82** [comment]
> Thanks, that is very nice of you. I have grown used to how some people write their posts, so I don't necessarily react much to it, at least not from users I have read a lot of comment posts from. Mind you, I can sometimes lose the overview of how much is affected by some of the issues, so in those cases I do take note of any follow-up comments from other users.

@illfated 
The fact is, I have found that sometimes people misunderstand the way I approach them.
So they assume I'm aggressive, but I'm not, I'm not **"Aggressive"** in my answers, I'm **"Peremptory"** which is different..

Now I don't know if this word **"Peremptory"** in English has the same definition it has in Italian, but the Italian definition is this:

```
Peremptory
Explicit and energetic, which does not admit reply or discussion.
```

**Comment by Zoidwaffle** [comment]
Yep, I also learned more about the whole decay+respawn numbers today :grin:

So perhaps 5 mins is too much - but I still think 1m is too little - perhaps 2 or 3m would be an acceptable compromise? 1m just passes really fast if you murder a cluster of mobs at once or if you die and need to run back.

@Stifler82 : and of course the millisecond part was a "typo" type of error, pretty obvious, just had to mention it :wink:

**Comment by Stifler82** [comment]
> Yep, I also learned more about the whole decay+respawn numbers today 😁
> 
> So perhaps 5 mins is too much - but I still think 1m is too little - perhaps 2 or 3m would be an acceptable compromise? 1m just passes really fast if you murder a cluster of mobs at once or if you die and need to run back.
> 
> @Stifler82 : and of course the millisecond part was a "typo" type of error, pretty obvious, just had to mention it 😉

@Zoidwaffle 
No, no, it wasn't a "type" of error, I just got confused..
